mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py

```diff
@@ -144,6 +154,7 @@ def finish(self):
                     self.selected_resource.closeConnection()
                 if self.selected_resource.is_allocated:
                     self.selected_resource.release()
+                    self.selected_resource = None
```

Comment:
reason I did like this was that if close connection or release throw exception there is still one chance to try - better to try again than forget because could be that if close connection fails in some reason it doesnt release at all and it might take time when it is finally released..